rust/gui-client/src-frontend/components/GeneralSettingsPage.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from "react";
+import { Button, Label, ToggleSwitch } from "flowbite-react";
+import { GeneralSettingsViewModel } from "../generated/GeneralSettingsViewModel";
+import { ManagedToggleSwitch, ManagedTextInput } from "./ManagedInput";
+
+interface Props {
+  settings: GeneralSettingsViewModel | null;
+  saveSettings: (settings: GeneralSettingsViewModel) => void;
+  resetSettings: () => void;
+}
+
+export default function GeneralSettingsPage({
+  settings,
+  saveSettings,
+  resetSettings,
+}: Props) {
+  // Local settings can be edited without affecting the global state.
+  const [localSettings, setLocalSettings] = useState<GeneralSettingsViewModel>(
+    settings ?? {
+      start_minimized: true,
+      account_slug: "",
+      connect_on_start: false,
+      start_on_login: false,
+      account_slug_is_managed: false,
+      connect_on_start_is_managed: false,
+    }
+  );
+
+  useEffect(() => {
+    setLocalSettings(
+      settings ?? {
+        start_minimized: true,
+        account_slug: "",
+        connect_on_start: false,
+        start_on_login: false,
+        account_slug_is_managed: false,
+        connect_on_start_is_managed: false,
+      }
+    );
+  }, [settings]);
+
+  return (
+    <div className="container p-4">
+      <div className="pb-2">
+        <h2 className="text-xl font-semibold">General settings</h2>
+      </div>
+
+      <form
+        onSubmit={(e) => {
+          e.preventDefault();
+          saveSettings(localSettings);
+        }}
+        className="max-w flex flex-col gap-2"
+      >
+        <div className="flex flex-row gap-2 items-center">
+          <ToggleSwitch
+            name="start_minimized"
+            id="start-minimized-input"
+            checked={localSettings.start_minimized}
+            onChange={(e) =>
+              setLocalSettings({
+                ...localSettings,
+                start_minimized: e,
+              })
+            }
+          />
+          <Label className="text-neutral-600" htmlFor="start-minimized-input">
+            Start minimized
+          </Label>
+        </div>
+
+        <div className="flex flex-row gap-2 items-center">
+          <ToggleSwitch
+            name="start_on_login"
+            id="start-on-login-input"
+            checked={localSettings.start_on_login}
+            onChange={(e) =>
+              setLocalSettings({
+                ...localSettings,
+                start_on_login: e,
+              })
+            }
+          />
+          <Label className="text-neutral-600" htmlFor="start-on-login-input">
+            Start on Login
+          </Label>
+        </div>
```

Comment:
Typically checkboxes are used on desktop platforms no? If these are checkboxes (like on macOS) then I think keeping the control in front of the label makes sense.

But if they're toggles then it might make sense to place the label first, and then the control after, right aligned, like so:

```suggestion
        <div className="flex flex-row justify-between items-center">
          <Label className="text-neutral-600" htmlFor="start-on-login-input">
            Start on Login
          </Label>
          <ToggleSwitch
            name="start_on_login"
            id="start-on-login-input"
            checked={localSettings.start_on_login}
            onChange={(e) =>
              setLocalSettings({
                ...localSettings,
                start_on_login: e,
              })
            }
          />
        </div>
```

Reply:
I used the switch because it was easier to implement the change as it directly gives me a boolean value. I thought those switches look nicer than the checkboxes.